src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +144,35 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
+    name = metadata_plugin.__class__.__name__
   return CallCredentials(
       _low.call_credentials_metadata_plugin(metadata_plugin, name))
 
+def google_oauth_call_credentials(credentials):
```

Comment:
+1 to having a  factory method to create call credentials  object  from   the credential object from the auth library, that's good for usability. Same as above, you probably don't want google_jwt_call_credentials factory method.

e.g. in C#,   the class that provides wrappers around  googleCredentials is called GoogleGrpcCredentials. C# supports extension methods, so i can actually do 

googleCredentials.ToCallCredentials()   and I obtain   a grpc-specific callCredentials from the original googleCredentials, but in python we can achieve the same using a factory method (basically what you are doing here, but I am wondering about the naming).

https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Auth/GoogleGrpcCredentials.cs